File: Task_1/c/csv_writer.c
/*
 * CSV Writer for C Matrix Multiplication Benchmark
 */
#include <stdio.h>
#include <string.h>
#include "csv_writer.h"

/**
 * Write benchmark results to CSV file
 */
void write_benchmark_to_csv(const char* language, int* sizes, int num_sizes, 
                           double** all_times, double** all_memory, 
                           int num_runs, const char* csv_file) {
    
    if (csv_file == NULL) return;
    
    FILE* file = fopen(csv_file, "a");
    if (file == NULL) {
        fprintf(stderr, "Warning: Could not open CSV file for writing\n");
        return;
    }
    
    // Write data for each size and run
    for (int s = 0; s < num_sizes; s++) {
        int size = sizes[s];
        double* times = all_times[s];
        double* memory = all_memory[s];
        
        for (int run = 0; run < num_runs; run++) {
            fprintf(file, "%s,%d,%d,%.6f,%.2f\n",
                    language, size, run + 1, times[run], memory[run]);
        }
    }
    
    fclose(file);
}

File: Task_1/java/CSVWriter.java
/**
 * CSV Writer for Java Matrix Multiplication Benchmark
 */
import java.io.*;

public class CSVWriter {
    
    /**
     * Write benchmark results to CSV file
     */
    public static void writeBenchmarkToCSV(String language, int[] sizes, 
            double[][] allTimes, double[][] allMemory, String csvFile) {
        
        if (csvFile == null) return;
        
        try (FileWriter fw = new FileWriter(csvFile, true);
             PrintWriter writer = new PrintWriter(fw)) {
            
            for (int s = 0; s < sizes.length; s++) {
                int size = sizes[s];
                double[] times = allTimes[s];
                double[] memory = allMemory[s];
                
                // Write each run as a separate row
                for (int run = 0; run < times.length; run++) {
                    writer.printf("%s,%d,%d,%.6f,%.2f%n",
                        language, size, run + 1, times[run], memory[run]);
                }
            }
            
        } catch (IOException e) {
            System.err.println("Warning: Could not write to CSV file: " + e.getMessage());
        }
    }
}